dify/docs/all.dify.how_to_guide.md

```diff
@@ -0,0 +1,170 @@
+<!-- toc -->
+
+- [Dify Chatflow with External Knowledge API](#dify-chatflow-with-external-knowledge-api)
+  * [Prerequisites](#prerequisites)
+  * [Step 1: Create Dify Application](#step-1-create-dify-application)
+  * [Step 2: Configure Basic Chatflow](#step-2-configure-basic-chatflow)
+    + [Start Node Configuration](#start-node-configuration)
+    + [Add LLM Node](#add-llm-node)
+    + [Connect and Test](#connect-and-test)
+  * [Step 3: Configure External Knowledge](#step-3-configure-external-knowledge)
+  * [Step 4: Add Knowledge to Chatflow](#step-4-add-knowledge-to-chatflow)
+    + [Add Knowledge Retrieval Node](#add-knowledge-retrieval-node)
+    + [Update LLM System Prompt](#update-llm-system-prompt)
+    + [Update User](#update-user)
+    + [Final Flow](#final-flow)
+  * [Step 5: Test and Optimize](#step-5-test-and-optimize)
+    + [Test Knowledge Integration](#test-knowledge-integration)
+    + [Optimization Options](#optimization-options)
+  * [Troubleshooting](#troubleshooting)
+    + [API Connection Issues](#api-connection-issues)
+    + [Poor Response Quality](#poor-response-quality)
+    + [Performance Issues](#performance-issues)
+
+<!-- tocstop -->
+
+# Dify Chatflow with External Knowledge API
+
+Step-by-step guide to create a Dify chatflow that uses your Weaviate knowledge
+base for intelligent document retrieval.
+
+## Prerequisites
+
+- Dify instance running (local or cloud)
+- Weaviate + Ollama setup completed
+- Documents processed into Weaviate collection
+- API service running on `http://localhost:2001`
+
+## Step 1: Create Dify Application
+
+1. Navigate to your Dify instance (`http://localhost/` for local)
+2. Click "Create Application" → "Chatflow"
```

Comment:
Can Dify be configured through code?

We want to push a button and the entire system is built.